ftplugin/markdown.vim

```diff
@@ -795,6 +799,7 @@ if !get(g:, 'vim_markdown_no_default_key_mappings', 0)
     call <sid>MapNotHasMapTo('<C-c>', 'Markdown_Checkbox', 'n')
     call <sid>MapNotHasMapTo('<TAB>', 'Markdown_Fold', 'n')
     call <sid>MapNotHasMapTo('<TAB>', 'Markdown_Jump', 'i')
+    call <sid>MapNotHasMapTo('<S-TAB>', 'Markdown_DeindentListItem', 'ni')
```

Comment:
It can be allowed, I suppose, if one wants to move an item higher in the hierarchy. Of course, the converse––<tab> to indent in normal mode––is not implemented yet.

Reply:
The intention with the tab (de)indenting is to have a convenient way to shift new bullets while editing, so that you don't have so switch modes and '<</>>'. This makes it so indentation is correct on newlines within bullet paragraphs, without having to adjust it afterwards, and it's nice to have it look right immediately. 

I see how it is useful with ordered lists though, as you always need it to re-enumerate whenever a bullet is shifted, but I think that would be better implemented through the nvim_buf_attach(on_lines=...) functionality. Same goes for the paste bindings you made I think now. With that you could edit the text in any way and have it work as expected.
This is an endeavour though, so while that isn't done I think it should just be kept simple. Despite, it's not really good to overwrite these basic normal mode bindings as they are very often changed by users. I've already done it with the folding, but I want to remove that as well.

Reply:
I did consider using `nvim_buf_attach` once, but it's triggered for every edit made. So, there's a performance consideration for low-end systems (perhaps, I am being pedantic here). For the same reason, I changed the `vim.on_key` call as well. I actually noticed perceptible improvement in the cursor-movement smoothness thereafter.

Reply:
Yes I was worried about that with vim.on_key, didn't know you could do it how you made it, much nicer now.
I see what you're saying with attach, idk if it triggers on every key in insert mode, but if it does it might be bad. I think maybe on_key locks the buffer while it processes the function call though, while attach wouldn't suffer from that I hope.